File: src/PcIfy.Server/Helpers/FileTypeHelper.cs
using PcIfy.Server.Constants;
using PcIfy.Server.DTOs.Files;

namespace PcIfy.Server.Helpers;

public static class FileTypeHelper
{
    public static FileType GetFileType(string extension)
    {
        if (MediaTypes.IsVideo(extension)) return FileType.Video;
        if (MediaTypes.IsImage(extension)) return FileType.Image;
        if (MediaTypes.IsAudio(extension)) return FileType.Audio;

        return extension.ToLowerInvariant() switch
        {
            ".pdf" or ".txt" or ".doc" or ".docx" or ".xls" or ".xlsx" => FileType.Document,
            ".zip" or ".rar" or ".7z" or ".tar" or ".gz"               => FileType.Archive,
            _                                                            => FileType.Unknown,
        };
    }
}

File: src/PcIfy.Server.Flutter/lib/core/utils/file_type_helper.dart
import '../constants/media_types.dart';

enum FileType { unknown, folder, video, image, audio, document, archive }

abstract final class FileTypeHelper {
  static const _docExts = {'pdf', 'txt', 'doc', 'docx', 'xls', 'xlsx'};
  static const _archiveExts = {'zip', 'rar', '7z', 'tar', 'gz'};

  static FileType fromExtension(String extension) {
    final ext = extension.toLowerCase().replaceFirst('.', '');
    if (MediaTypes.isVideo(ext)) return FileType.video;
    if (MediaTypes.isImage(ext)) return FileType.image;
    if (MediaTypes.isAudio(ext)) return FileType.audio;
    if (_docExts.contains(ext)) return FileType.document;
    if (_archiveExts.contains(ext)) return FileType.archive;
    return FileType.unknown;
  }
}
